precompiles/utils/src/lib.rs

```diff
@@ -181,13 +181,13 @@ where
 		}
 
 		// Dispatch call.
-		// It may be possible to not record gas cost if the call returnes Pays::No.
+		// It may be possible to not record gas cost if the call returns Pays::No.
 		// However while Substrate handle checking weight while not making the sender pay for it,
 		// the EVM doesn't. It seems this safer to always record the costs to avoid unmetered
 		// computations.
 		let used_weight = call
 			.dispatch(origin)
-			.map_err(|_| error("dispatched call failed"))?
+			.map_err(|e| error(alloc::format!("Dispatched call failed with error: {:?}", e)))?
```

Comment:
This change preserves the error propagation added in https://github.com/PureStake/moonbeam/pull/677 to facilitate debugging for users whenever the dispatched call fails